packages/message-manager/src/AbstractMessageManager.ts

```diff
@@ -242,6 +247,25 @@ export abstract class AbstractMessageManager<
     this.setMessageStatus(messageId, 'signed');
   }
 
+  /**
+   * Sets a EncryptionPublicKey status to 'received' via a call to this.setMsgStatus and
+   * updates that EncryptionPublicKey in this.messages by adding the raw data of request
+   * to the EncryptionPublicKey.
+   *
+   * @param messageId - The id of the Message to sign.
+   * @param rawSig - The encryption public key of the request.
+   */
+  setMessageStatusReceived(messageId: string, rawSig: string) {
```

Comment:
This follows the existing pattern on `AbstractMessageManager` in which methods are added for each status change.